core/state_processor.go

```diff
@@ -134,16 +129,17 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 // ApplyTransactionWithEVM attempts to apply a transaction to the given state database
 // and uses the input parameters for its environment similar to ApplyTransaction. However,
 // this method takes an already created EVM instance as input.
-func ApplyTransactionWithEVM(msg *Message, gp *GasPool, statedb *state.StateDB, blockNumber *big.Int, blockHash common.Hash, tx *types.Transaction, usedGas *uint64, evm *vm.EVM) (receipt *types.Receipt, err error) {
+func ApplyTransactionWithEVM(msg *Message, gp *GasPool, statedb *state.StateDB, tx *types.Transaction, usedGas *uint64, evm *vm.EVM) (receipt *types.Receipt, err error) {
 	if hooks := evm.Config.Tracer; hooks != nil {
 		if hooks.OnTxStart != nil {
 			hooks.OnTxStart(evm.GetVMContext(), tx, msg.From)
 		}
 		if hooks.OnTxEnd != nil {
+			// TODO (rjl493456442) the receipt only contain consensus fields,
+			// should we derive the others here?
```

Comment:
Yes please. I can spot quite a few of these fields being used here https://github.com/streamingfast/go-ethereum/blob/9bf4d2482fb4fc3ea208fa55884c63d56a6858e5/eth/tracers/firehose.go#L576 including TransactionIndex, GasUsed, BlobGasUsed, BlobGasPrice.